src/plugins/index_pattern_editor/public/components/index_pattern_editor_flyout_content.tsx

```diff
@@ -0,0 +1,372 @@
+
+  const indexPatternTypeSelect = showIndexPatternTypeSelect() ? (
+    <>
+      <EuiFlexGroup>
+        <EuiFlexItem>
+          <TypeField
+            onChange={(newType) => {
+              form.setFieldValue('title', '');
+              form.setFieldValue('timestampField', '');
+              if (newType === INDEX_PATTERN_TYPE.ROLLUP) {
+                form.setFieldValue('allowHidden', false);
+              }
+            }}
+          />
+        </EuiFlexItem>
+      </EuiFlexGroup>
+      {type === INDEX_PATTERN_TYPE.ROLLUP ? (
+        <EuiFlexGroup>
+          <EuiFlexItem>
+            <RollupBetaWarning />
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      ) : (
+        <></>
+      )}
+    </>
+  ) : (
+    <></>
+  );
+
+  return (
+    <EmptyPrompts
+      onCancel={onCancel}
+      allSources={allSources}
+      hasExistingIndexPatterns={!!existingIndexPatterns.length}
+      loadSources={loadSources}
+    >
+      <FlyoutPanels.Group flyoutClassName={'indexPatternEditorFlyout'} maxWidth={1180}>
+        <FlyoutPanels.Item className="fieldEditor__mainFlyoutPanel" border="right">
+          <EuiTitle data-test-subj="flyoutTitle">
+            <h2>Create index pattern</h2>
+          </EuiTitle>
+          <Form form={form} className="indexPatternEditor__form">
+            {indexPatternTypeSelect}
+            <EuiSpacer size="m" />
+            <EuiFlexGroup>
+              <EuiFlexItem>
+                <TitleField
+                  isRollup={form.getFields().type?.value === INDEX_PATTERN_TYPE.ROLLUP}
+                  existingIndexPatterns={existingIndexPatterns}
+                  refreshMatchedIndices={reloadMatchedIndices}
+                  matchedIndices={matchedIndices.exactMatchedIndices}
+                  rollupIndicesCapabilities={rollupIndicesCapabilities}
+                />
+              </EuiFlexItem>
+            </EuiFlexGroup>
+            <EuiSpacer size="m" />
+            <EuiFlexGroup>
+              <EuiFlexItem>
+                <TimestampField
+                  options={timestampFieldOptions}
+                  isLoadingOptions={isLoadingTimestampFields}
+                  isExistingIndexPattern={existingIndexPatterns.includes(title)}
+                  isLoadingMatchedIndices={isLoadingMatchedIndices}
+                  hasMatchedIndices={!!matchedIndices.exactMatchedIndices.length}
+                />
+              </EuiFlexItem>
+            </EuiFlexGroup>
+            <AdvancedParamsContent disableAllowHidden={type === INDEX_PATTERN_TYPE.ROLLUP} />
+          </Form>
+          <Footer
+            onCancel={onCancel}
+            onSubmit={() => form.submit()}
+            submitDisabled={form.isSubmitted && !form.isValid}
+          />
+        </FlyoutPanels.Item>
+        <FlyoutPanels.Item>
+          {isLoadingSources ? (
+            <></>
+          ) : (
+            <PreviewPanel
+              type={type as INDEX_PATTERN_TYPE}
```

Comment:
No need to cast as it is already `INDEX_PATTERN_TYPE`

Reply:
addressed